src/Utils/obiDialogParser.ts

```diff
@@ -369,6 +487,11 @@ export class ObiDialogParser {
             throw new Error('HTTP requests require url')
         }
         // TODO(thpar) : revisit logic for this.
+        // Note that we cannot do this 100% correctly in the current implementation, since actions (scorer steps)
+        // from a given dialog tree node $Y may be added to rounds from the previous dialog tree node $X if $Y does
+        // not have an extractor step, but we are calling this method during the handling of $X.
+        // To handle this 100% correctly, we'd need to do a multi-pass traversal of the dialog tree or build up a
+        // second tree-like representation of extractor and scorer steps.
```

Comment:
Is there any way to catch this if it happens?  Concerned about a silent failure.

Reply:
I think the import will fail if this happens, since we'd have a round with a terminal action followed by another scorer step.